File: Dopamine.FullPlayerModule/ViewModels/MainScreenViewModel.vb
Imports Dopamine.Common.Presentation.Views
Imports Dopamine.ControlsModule.Views
Imports Dopamine.Core.Prism
Imports Microsoft.Practices.Prism.Commands
Imports Microsoft.Practices.Prism.Mvvm
Imports Microsoft.Practices.Prism.Regions

Namespace ViewModels
    Public Class MainScreenViewModel
        Inherits BindableBase

#Region "Variables"
        Private ReadOnly mRegionManager As IRegionManager
        Private mPreviousIndex As Integer = 0
        Private mSubMenuSlideInFrom As Integer
        Private mSearchSlideInFrom As Integer
        Private mContentSlideInFrom As Integer
#End Region

#Region "Properties"
        Public Property NavigateBetweenMainCommand As DelegateCommand(Of String)

        Public Property SubMenuSlideInFrom() As Integer
            Get
                Return mSubMenuSlideInFrom
            End Get
            Set(ByVal value As Integer)
                SetProperty(Of Integer)(mSubMenuSlideInFrom, value)
            End Set
        End Property

        Public Property SearchSlideInFrom() As Integer
            Get
                Return mSearchSlideInFrom
            End Get
            Set(ByVal value As Integer)
                SetProperty(Of Integer)(mSearchSlideInFrom, value)
            End Set
        End Property

        Public Property ContentSlideInFrom() As Integer
            Get
                Return mContentSlideInFrom
            End Get
            Set(ByVal value As Integer)
                SetProperty(Of Integer)(mContentSlideInFrom, value)
            End Set
        End Property
#End Region

#Region "Construction"
        Public Sub New(iRegionManager As IRegionManager)
            Me.mRegionManager = iRegionManager

            Me.NavigateBetweenMainCommand = New DelegateCommand(Of String)(Sub(iIndex) Me.NavigateBetweenMain(iIndex))
            ApplicationCommands.NavigateBetweenMainCommand.RegisterCommand(Me.NavigateBetweenMainCommand)
            Me.SubMenuSlideInFrom = 30
            Me.SearchSlideInFrom = 30
            Me.ContentSlideInFrom = 30
        End Sub
#End Region

#Region "Private"
        Private Sub NavigateBetweenMain(iIndex As String)

            If String.IsNullOrWhiteSpace(iIndex) Then Return

            Dim index As Integer = 0

            Integer.TryParse(iIndex, index)

            If index = 0 Then Return

            Me.SubMenuSlideInFrom = If(index <= mPreviousIndex, 0, 30)
            Me.SearchSlideInFrom = If(index <= mPreviousIndex, -30, 30)
            Me.ContentSlideInFrom = If(index <= mPreviousIndex, -30, 30)

            mPreviousIndex = index

            If index = 2 Then
                ' Settings
                Me.mRegionManager.RequestNavigate(RegionNames.ContentRegion, GetType(Dopamine.SettingsModule.Views.Settings).FullName)
                Me.mRegionManager.RequestNavigate(RegionNames.SubMenuRegion, GetType(Dopamine.SettingsModule.Views.SettingsSubMenu).FullName)
                Me.mRegionManager.RequestNavigate(RegionNames.FullPlayerSearchRegion, GetType(Empty).FullName)
            ElseIf index = 3 Then
                ' Information
                Me.mRegionManager.RequestNavigate(RegionNames.ContentRegion, GetType(Dopamine.InformationModule.Views.Information).FullName)
                Me.mRegionManager.RequestNavigate(RegionNames.SubMenuRegion, GetType(Dopamine.InformationModule.Views.InformationSubMenu).FullName)
                Me.mRegionManager.RequestNavigate(RegionNames.FullPlayerSearchRegion, GetType(Empty).FullName)
            Else
                ' Collection
                Me.mRegionManager.RequestNavigate(RegionNames.ContentRegion, GetType(Dopamine.CollectionModule.Views.Collection).FullName)
                Me.mRegionManager.RequestNavigate(RegionNames.SubMenuRegion, GetType(Dopamine.CollectionModule.Views.CollectionSubMenu).FullName)
                Me.mRegionManager.RequestNavigate(RegionNames.FullPlayerSearchRegion, GetType(SearchControl).FullName)
            End If
        End Sub
#End Region
    End Class
End Namespace

File: Dopamine.FullPlayerModule/ViewModels/MainScreenViewModel.cs
using Dopamine.Common.Presentation.Views;
using Dopamine.ControlsModule.Views;
using Dopamine.Core.Prism;
using Microsoft.Practices.Prism.Commands;
using Microsoft.Practices.Prism.Mvvm;
using Microsoft.Practices.Prism.Regions;

namespace Dopamine.FullPlayerModule.ViewModels
{
    public class MainScreenViewModel : BindableBase
    {
        #region Variables
        private readonly IRegionManager regionManager;
        private int previousIndex = 0;
        private int subMenuSlideInFrom;
        private int searchSlideInFrom;
        private int contentSlideInFrom;
        #endregion

        #region Commands
        public DelegateCommand<string> NavigateBetweenMainCommand { get; set; }
        #endregion

        #region Properties
        public int SubMenuSlideInFrom
        {
            get { return this.subMenuSlideInFrom; }
            set { SetProperty<int>(ref this.subMenuSlideInFrom, value); }
        }

        public int SearchSlideInFrom
        {
            get { return this.searchSlideInFrom; }
            set { SetProperty<int>(ref this.searchSlideInFrom, value); }
        }

        public int ContentSlideInFrom
        {
            get { return this.contentSlideInFrom; }
            set { SetProperty<int>(ref this.contentSlideInFrom, value); }
        }
        #endregion

        #region Construction
        public MainScreenViewModel(IRegionManager regionManager)
        {
            this.regionManager = regionManager;

            this.NavigateBetweenMainCommand = new DelegateCommand<string>((index) => this.NavigateBetweenMain(index));
            ApplicationCommands.NavigateBetweenMainCommand.RegisterCommand(this.NavigateBetweenMainCommand);
            this.SubMenuSlideInFrom = 30;
            this.SearchSlideInFrom = 30;
            this.ContentSlideInFrom = 30;
        }
        #endregion

        #region Private
        private void NavigateBetweenMain(string indexString)
        {
            if (string.IsNullOrWhiteSpace(indexString))
                return;

            int index = 0;

            int.TryParse(indexString, out index);

            if (index == 0)
                return;

            this.SubMenuSlideInFrom = index <= this.previousIndex ? 0 : 30;
            this.SearchSlideInFrom = index <= this.previousIndex ? -30 : 30;
            this.ContentSlideInFrom = index <= this.previousIndex ? -30 : 30;

            this.previousIndex = index;

            if (index == 2)
            {
                // Settings
                this.regionManager.RequestNavigate(RegionNames.ContentRegion, typeof(SettingsModule.Views.Settings).FullName);
                this.regionManager.RequestNavigate(RegionNames.SubMenuRegion, typeof(SettingsModule.Views.SettingsSubMenu).FullName);
                this.regionManager.RequestNavigate(RegionNames.FullPlayerSearchRegion, typeof(Empty).FullName);
            }
            else if (index == 3)
            {
                // Information
                this.regionManager.RequestNavigate(RegionNames.ContentRegion, typeof(InformationModule.Views.Information).FullName);
                this.regionManager.RequestNavigate(RegionNames.SubMenuRegion, typeof(InformationModule.Views.InformationSubMenu).FullName);
                this.regionManager.RequestNavigate(RegionNames.FullPlayerSearchRegion, typeof(Empty).FullName);
            }
            else
            {
                // Collection
                this.regionManager.RequestNavigate(RegionNames.ContentRegion, typeof(CollectionModule.Views.Collection).FullName);
                this.regionManager.RequestNavigate(RegionNames.SubMenuRegion, typeof(CollectionModule.Views.CollectionSubMenu).FullName);
                this.regionManager.RequestNavigate(RegionNames.FullPlayerSearchRegion, typeof(SearchControl).FullName);
            }
        }
        #endregion
    }
}
